EvidenceApi/V1/UseCase/CreateResidentReferenceIdUseCase.cs

```diff
@@ -0,0 +1,56 @@
+using System.Collections.Generic;
+using System.Linq;
+using EvidenceApi.V1.Domain;
+using EvidenceApi.V1.Gateways.Interfaces;
+using EvidenceApi.V1.Infrastructure.Interfaces;
+using EvidenceApi.V1.UseCase.Interfaces;
+
+namespace EvidenceApi.V1.UseCase
+{
+    public class CreateResidentReferenceIdUseCase : ICreateResidentReferenceIdUseCase
```

Comment:
I think the `ResidentReferenceId` is not always created by this use case as sometimes it will return the existing one so:
```suggestion
    public class FindOrCreateResidentReferenceIdUseCase : ICreateResidentReferenceIdUseCase
```

Reply:
Good point! Renamed